mesh_to_sampled_point_cloud/src/pcl_mesh_sampling_extract.cpp

```diff
@@ -160,6 +160,20 @@ std::shared_ptr<pcl::PointCloud<pcl::PointXYZ>> PCLMeshSamplingExtract::getSampl
 }
 
 
+std::shared_ptr<pcl::PointCloud<pcl::PointNormal>> PCLMeshSamplingExtract::getSampledPointCloudXYZN()
+{
+  if (!write_normals_)
+  {
+    std::cout << "Parameter \"write_normals\" must be set for calling this function" << std::endl;
+  }
```

Comment:
Can your program continue if the point cloud normals aren't valid?
If not it makes no sense only to print something on the console that no one will see (raise an exception and document it on the function declaration that it may throw in that case).
And either way I would at least use a `ROS_ERROR` or `ROS_WARN` so it is better visible, will also be written to the logs and broadcasted.